Controllers/MemberController.cs

```
@@ -0,0 +1,151 @@
+            if (member == null)
+            {
+                return NotFound();
+            }
+
+            return View(member);
+        }
+
+        // GET: Member/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Member/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Name,DateOfRegistration,Email")] Member member)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Member.Add(member);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            return View(member);
+        }
+
+        // GET: Member/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var member = await _context.Member.FindAsync(id);
+            if (member == null)
+            {
+                return NotFound();
+            }
+            return View(member);
+        }
+
+        // POST: Member/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, [Bind("Id,Title")] Member member)
+        {
+            if (id != member.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    _context.Update(member);
+                    await _context.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!MemberExists(member.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
```

Can we throw the same error that we received (the `DbUpdateConcurrencyException`). We can do this with
```
catch (DbUpdateConcurrencyException exception) {
    ...
        throw exception;
    ...
}
```